pkg/controller/elasticsearch/reconcile/status.go

```diff
@@ -191,20 +194,23 @@ func (u *UpgradeReporter) Merge(other esv1.UpgradeOperation) esv1.UpgradeOperati
 	if u == nil {
 		return *upgradeOperation
 	}
-	nodes := make([]esv1.UpgradedNode, 0, len(u.nodes))
-	for _, node := range u.nodes {
-		nodes = append(nodes, esv1.UpgradedNode{
-			Name:      node.Name,
-			Predicate: node.Predicate,
-			Message:   node.Message,
-			Status:    node.Status,
+	var nodes []esv1.UpgradedNode
+	if len(u.nodes) != 0 {
+		nodes = make([]esv1.UpgradedNode, 0, len(u.nodes))
+		for _, node := range u.nodes {
+			nodes = append(nodes, esv1.UpgradedNode{
+				Name:      node.Name,
+				Predicate: node.Predicate,
+				Message:   node.Message,
+				Status:    node.Status,
+			})
+		}
+		// Sort for stable comparison
+		sort.Slice(nodes, func(i, j int) bool {
+			return nodes[i].Name < nodes[j].Name
 		})
 	}
-	// Sort for stable comparison
-	sort.Slice(nodes, func(i, j int) bool {
-		return nodes[i].Name < nodes[j].Name
-	})
-	if (u.nodes != nil && !reflect.DeepEqual(nodes, other.Nodes)) || upgradeOperation.Nodes == nil {
+	if !reflect.DeepEqual(nodes, other.Nodes) || upgradeOperation.LastUpdatedTime.IsZero() {
```

Comment:
I think we should also test if `u.nodes` is not `nil` There are some cases where the upgrade logic may not been invoked. In that cases we don't want to clear the current status, because the reporter has not been used at all. This is why the [current condition](https://github.com/elastic/cloud-on-k8s/blob/84b13c74d7a367122190b78feaa650ffa21424ed/pkg/controller/elasticsearch/reconcile/status.go#L207) is `u.nodes != nil && !reflect.DeepEqual(nodes, other.Nodes)`

Reply:
Makes sense.